### PR TITLE
feat: add exec command support for xlinectl lock

### PR DESCRIPTION
**Pull request by bhavik-goplani**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
To solve issue: #823 

* what changes does this pull request make?
Change the `xlinectl lock` command

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

**Comment by CrystalAnalyst** [comment]
The overall logic is clear and the testing part is okay , please squash them into 1 commits.

File: crates/xlinectl/src/command/lock.rs [hunk 1]
```diff
@@ -15,9 +17,29 @@ pub(crate) async fn execute(client: &mut Client, matches: &ArgMatches) -> Result
...
+            let (command, args) = exec_command_vec
+                .split_first()
+                .expect("Expected at least one exec command");
+            let output = StdCommand::new(command)
+                .args(args)
+                .output()
+                .expect("failed to execute command");
+            println!("{}", String::from_utf8_lossy(&output.stdout).trim());
+            should_wait_for_ctrl_c = false;
+        }
```
[the remark this answers is not in the document]
Comment (bhavik-goplani) [reply crates/xlinectl/src/command/lock.rs:36]: We could add the `println!("releasing the lock, ")` but I wanted to keep the exec arg implementation as close to etcdctl as possible and yes, it means that if the exec command is successful, we directly drop the lock without waiting for the user to press `ctrl c`

File: scripts/validation_test.sh [hunk 2]
```diff
@@ -6,6 +6,8 @@ source $DIR/log.sh
 QUICK_START="${DIR}/quick_start.sh"
 ETCDCTL="docker exec -i client etcdctl --endpoints=http://172.20.0.3:2379,http://172.20.0.4:2379"
 
+XLINECTL="${DIR}/../target/debug/xlinectl"
```
[the remark this answers is not in the document]
Comment (bhavik-goplani) [reply scripts/validation_test.sh:9]: Sounds good, I'll make that change. Thanks!

File: scripts/validation_test.sh [hunk 3]
```diff
@@ -272,6 +274,12 @@ lock_validation() {
     run "${ETCDCTL} lock mutex echo success"
     check_positive "success"
 
+    run "${XLINECTL} lock mutex echo success"
+    check_positive "success"
+
+    run "${XLINECTL} lock mutex ${XLINECTL} put key1 value1"
```
[the remark this answers is not in the document]
Comment (bhavik-goplani) [reply scripts/validation_test.sh:280]: I'll make the change

**Comment by CrystalAnalyst** [comment]
Hi, Bhavik! I really appreciate your passion and resilience. Please add some unit test, and squash your commits into one.

**Comment by bhavik-goplani** [comment]
@CrystalAnalyst Thanks for your feedback! I have made the final changes and added unit tests.

File: crates/xlinectl/src/command/lock.rs [hunk 4]
```diff
@@ -15,9 +17,79 @@ pub(crate) async fn execute(client: &mut Client, matches: &ArgMatches) -> Result
...
+                self.expected_output,
+                output
+            );
+        }
+    }
+
+    #[test]
+    fn test_execute_exec_command() {
+        let test_cases = vec![TestCase::new(vec!["echo", "success"], "success".to_owned())];
+
```
Comment (CrystalAnalyst) [inline crates/xlinectl/src/command/lock.rs:90]: You can add more test cases here, maybe 2 ~ 3 ?  When you finish, you can pass all the CI tests.

Comment (bhavik-goplani) [reply crates/xlinectl/src/command/lock.rs:95]: @CrystalAnalyst I have made the changes and squashed my commits as well

**Comment by bhavik-goplani** [comment]
@CrystalAnalyst Do you think I need to add more tests to hit the 75.5 % diff target?

**Review by CrystalAnalyst — APPROVED** [review]

**Pull request merged**